### DISCUSSION TITLE
synology?

### DISCUSSION THREAD

**Discussion by DennisLindblom71** [emmercm/igir#854] [General]
I would like to use this on my synology nas which should I run? I've tried them all but they haven't worked

**Comment by emmercm** [comment]
Hi @DennisLindblom71, I would need some more information from you:

- What are you trying to accomplish?
- What are the unexpected results?
- How are you running `igir` (i.e. over SSH, in a Docker container, etc.)? I'm not familiar with what the environment is for Synology NASes

**Comment by dxman** [comment]
Rather late to the party, but I just got done setting up Igir on my Synology NAS. I did it by installing the NPM package globally. I didn't have any luck with the Homebrew package, but I think that was more my fault than Igir's.

The initial scan is running as we speak, so it remains to be seen if any issues arise, but my intended workflow is to SSH into my NAS and use Igir to maintain my complete library in one directory, then create symlinks for a subset of games in my Romm library directory. From there, I can use Romm's Playnite extension to download ROMs to my PC to play.

EDIT: It looks like chdman is not detecting the presence of the installed SDL2 library. If I run chdman by itself, after setting LD_LIBRARY_PATH to the correct location, I can get that to work. But the errors persist when running Igir, even after setting that variable. It's no big deal for now though. It's late, and all my non-CHD files are processing correctly, so I'll revisit my approach later.

**Reply by dxman** [reply]
Update: Got it working by creating a symlink:

`sudo ln -s $HOMEBREW_PREFIX/opt/sdl2/lib/libSDL2-2.0.so.0 /usr/lib/libSDL2-2.0.so.0`

Working on tweaking a set of scripts for my purposes, so I might share them once they're complete, if anyone's interested.

**Reply by emmercm** [reply]
@dxman https://github.com/emmercm/chdman-js/pull/61 should obviate the need for that symlink.

**Reply by chrisdrackett** [reply]
I think this is running on the synology, i'm working on the same thing

**Reply by dxman** [reply]
> I think this is running on the synology, i'm working on the same thing

Yeah, it was working for me too, last time I checked. Sorry for not replying earlier to confirm. Just juggling a lot of different projects at the moment. XD

**Comment by chrisdrackett** [comment]
I'm also working on running on my synology as running on my mac with SMB shares always leads to disconnects for one reason or another. So far I've installed Node22 from the package center and homebrew so I can install other requirements. I'll try and keep notes so when its working I can outline somewhere to share
